library/Fission/User/Mutate.hs

```diff
@@ -71,6 +74,29 @@ create' username password email herokuUUID = do
       logInfo $ "Inserted user " <> display uID
       return $ Right uID
 
+updatePassword :: MonadRIO    cfg m
+               => MonadSelda      m
+               => HasLogFunc  cfg
+               => ID User
+               -> User.Password
+               -> m (Either Error.Create User.Password)
+updatePassword userID (User.Password maybePass) = do
+  password <- defaultPassword maybePass
+  hashPassword' password >>= \case
+    Left err -> return $ Left err
+    Right secretDigest -> do
+      update Table.users 
+        (\user -> user ! #_userID .== literal userID)
+        (\user -> user `with` [#_secretDigest := literal secretDigest])
+      logInfo $ "Updated password for user " <> display userID
+      return . Right . User.Password $ Just password
+
+defaultPassword :: MonadRIO cfg m
+                => Maybe Text
+                -> m (Text)
```

Comment:
💅

* No need for the brackets around `(Text)`
* No need for the brackets around `(Just pw)` in this case
* This doesn't need to be full `MonadRIO`; `MonadIO` will do
  * `MonadRIO` is for when you need to pass around some config

```haskell
defaultPassword :: MonadIO m => Maybe Text -> m Text
defaultPassword Just pw = return pw
defaultPassword Nothing = liftIO $ Random.text 200
```

Reply:
Good point on 1 & 3. Compiler freaks out without brackets around `Just pw` though: `Equations for ‘defaultPassword’ have different numbers of arguments`